internal/sbi/processor/nf_discovery.go

```diff
@@ -143,30 +143,31 @@ func (p *Processor) NFDiscoveryProcedure(c *gin.Context, queryParameters url.Val
 
 	// handle ipv4 & ipv6
 	if queryParameters["target-nf-type"][0] == "BSF" {
-		for i, nfProfile := range nfProfilesStruct {
+		for i := 0; i < len(nfProfilesStruct); i++ {
```

Comment:
Why not use range-based loop?

Also, using `i` and `j` in such a complex function will make the code hard to read. Try to rename these variables to make them sound more reasonable.😉